python/ray/serve/controller.py

```diff
@@ -119,9 +121,32 @@ async def run_control_loop(self) -> None:
                     self.backend_state.update()
                 except Exception as e:
                     logger.error(f"Exception updating backend state: {e}")
-
+            self._put_serve_snapshot()
             await asyncio.sleep(CONTROL_LOOP_PERIOD_S)
 
+    def _put_serve_snapshot(self) -> None:
+        val = []
+        for name, (backend_info,
+                   route_prefix) in self.list_deployments().items():
+            deployment_info = {}
+            deployment_info["name"] = name
+            deployment_info["namespace"] = ray.get_runtime_context().namespace
+            deployment_info["ray_job_id"] = (
+                "None" if backend_info.deployer_job_id is None else
+                backend_info.deployer_job_id.hex())
+            deployment_info[
+                "class_name"] = backend_info.replica_config.func_or_class_name
+            deployment_info["version"] = backend_info.version or "Unversioned"
+            # TODO(architkulkarni): When we add the feature to allow
+            # deployments with no HTTP route, update the below line.
+            # Or refactor the route_prefix logic in the Deployment class now.
+            deployment_info["http_route"] = route_prefix or f"/{name}"
+            deployment_info["status"] = "RUNNING"
+            deployment_info["start_time"] = "TODO"
+            deployment_info["end_time"] = ""
+            val.append(deployment_info)
+        self.kv_store.put(SNAPSHOT_KEY, json.dumps(val))
```

Comment:
Could we actually change this to be like the other fields: `actors` and `jobs` in which 
its `{"identifier": value}`
In this case it would be `{"deployment_name": value}`.

Reply:
also @architkulkarni  start_time / end_time are planned to be unix timestamps in the future correct?

I think it will be better if we use `0` for start_time and end_time right now instead of a string so we don't have to change the type in the end.

The jobs and actors use `0` for when end_time is not known

Reply:
This all sounds good, I had the same thought regarding the first point.  I'll include it in the followup PR if that's okay.  As for start/end times, it hasn't been implemented yet so we have freedom to choose there--your suggestion sounds good to me.